"Additional property 'useBootstrapper' is not allowed" when building from source

I'm trying to build from source on an M1 Mac (2021 Macbook Pro, macOS Monterey 12.3.1). In `make build-release` I am getting:

```
cargo tauri build
`tauri.conf.json` error on `tauri > bundle > deb`: Additional property 'useBootstrapper' is not allowed
`tauri.conf.json` error on `tauri > bundle > macOS`: Additional property 'useBootstrapper' is not allowed
make: *** [build-release] Error 1
```

Thanks for making this! I don't mind installing from the release, I just figured since this is an M1 mac it might be faster if built from source.

--- a/crates/shared/src/config.rs
+++ b/crates/shared/src/config.rs
@@ -77,7 +77,7 @@ impl Default for UserSettings {
             // Not used at the moment
             run_wizard: false,
             allow_list: Vec::new(),
-            block_list: vec!["web.archive.org".to_string()]
+            block_list: vec!["web.archive.org".to_string()],
         }
     }
 }
@@ -122,11 +122,14 @@ impl Config {
             let lens = Lens {
                 version: "1".to_string(),
                 name: "wiki".to_string(),
-                description: Some("Search through official user-supported wikis for knowledge, games, and more.".to_string()),
+                description: Some(
+                    "Search through official user-supported wikis for knowledge, games, and more."
+                        .to_string(),
+                ),
                 domains: vec![
                     "en.wikipedia.org".to_string(),
                     "oldschool.runescape.wiki".to_string(),
-                    "wiki.factorio.com".to_string()
+                    "wiki.factorio.com".to_string(),
                 ],
                 urls: Vec::new(),
             };
@@ -135,7 +138,7 @@ impl Config {
                 Self::lenses_dir().join("wiki.ron"),
                 ron::ser::to_string_pretty(&lens, Default::default()).unwrap(),
             )
-                .expect("Unable to save default lens file.");
+            .expect("Unable to save default lens file.");
         }
 
         Ok(lenses)

--- a/crates/spyglass/src/api/route.rs
+++ b/crates/spyglass/src/api/route.rs
@@ -130,7 +130,6 @@ pub async fn toggle_pause(state: AppState) -> jsonrpc_core::Result<AppStatus> {
         let app_state = &state.app_state;
         let mut paused_status = app_state.get_mut("paused").unwrap();
 
-        log::info!("dlafjadkl");
         let current_status = paused_status.to_string() == "true";
         let updated_status = !current_status;
         *paused_status = updated_status.to_string();

--- a/crates/tauri/src/cmd.rs
+++ b/crates/tauri/src/cmd.rs
@@ -1,0 +1,80 @@
+use shared::{request, response};
+use tauri::{LogicalSize, Size, State};
+
+use crate::_hide_window;
+use crate::constants;
+use crate::rpc;
+
+#[tauri::command]
+pub async fn escape(window: tauri::Window) -> Result<(), String> {
+    _hide_window(&window);
+    Ok(())
+}
+
+#[tauri::command]
+pub async fn open_result(_: tauri::Window, url: &str) -> Result<(), String> {
+    open::that(url).unwrap();
+    Ok(())
+}
+
+#[tauri::command]
+pub fn resize_window(window: tauri::Window, height: f64) {
+    window
+        .set_size(Size::Logical(LogicalSize {
+            width: constants::INPUT_WIDTH,
+            height,
+        }))
+        .unwrap();
+}
+
+#[tauri::command]
+pub async fn search_docs<'r>(
+    _: tauri::Window,
+    rpc: State<'r, rpc::RpcClient>,
+    lenses: Vec<String>,
+    query: &str,
+) -> Result<Vec<response::SearchResult>, String> {
+    let data = request::SearchParam {
+        lenses,
+        query: query.to_string(),
+    };
+
+    match rpc
+        .client
+        .call_method::<(request::SearchParam,), response::SearchResults>("search_docs", "", (data,))
+        .await
+    {
+        Ok(resp) => Ok(resp.results.to_vec()),
+        Err(err) => {
+            log::error!("{}", err);
+            Ok(Vec::new())
+        }
+    }
+}
+
+#[tauri::command]
+pub async fn search_lenses<'r>(
+    _: tauri::Window,
+    rpc: State<'r, rpc::RpcClient>,
+    query: &str,
+) -> Result<Vec<response::LensResult>, String> {
+    let data = request::SearchLensesParam {
+        query: query.to_string(),
+    };
+
+    match rpc
+        .client
+        .call_method::<(request::SearchLensesParam,), response::SearchLensesResp>(
+            "search_lenses",
+            "",
+            (data,),
+        )
+        .await
+    {
+        Ok(resp) => Ok(resp.results.to_vec()),
+        Err(err) => {
+            log::error!("{}", err);
+            Ok(Vec::new())
+        }
+    }
+}

--- a/crates/tauri/src/cmd.rs
+++ b/crates/tauri/src/cmd.rs
@@ -1,13 +1,11 @@
 use shared::{request, response};
-use tauri::{LogicalSize, Size, State};
+use tauri::State;
 
-use crate::_hide_window;
-use crate::constants;
-use crate::rpc;
+use crate::{rpc, window};
 
 #[tauri::command]
 pub async fn escape(window: tauri::Window) -> Result<(), String> {
-    _hide_window(&window);
+    window::hide_window(&window);
     Ok(())
 }
 
@@ -19,12 +17,7 @@ pub async fn open_result(_: tauri::Window, url: &str) -> Result<(), String> {
 
 #[tauri::command]
 pub fn resize_window(window: tauri::Window, height: f64) {
-    window
-        .set_size(Size::Logical(LogicalSize {
-            width: constants::INPUT_WIDTH,
-            height,
-        }))
-        .unwrap();
+    window::resize_window(&window, height);
 }
 
 #[tauri::command]

--- a/crates/tauri/src/constants.rs
+++ b/crates/tauri/src/constants.rs
@@ -1,0 +1,5 @@
+pub const INPUT_WIDTH: f64 = 640.0;
+pub const INPUT_HEIGHT: f64 = 80.0;
+pub const INPUT_Y: f64 = 128.0;
+
+pub const SHORTCUT: &str = "CmdOrCtrl+Shift+/";

--- a/crates/tauri/src/main.rs
+++ b/crates/tauri/src/main.rs
@@ -5,8 +5,7 @@
 use jsonrpc_core::Value;
 use num_format::{Locale, ToFormattedString};
 use std::path::PathBuf;
-use tauri::api::process::Command;
-use tauri::{AppHandle, GlobalShortcutManager, Manager, SystemTray, SystemTrayEvent, Window};
+use tauri::{AppHandle, GlobalShortcutManager, Manager, SystemTray, SystemTrayEvent};
 
 use shared::config::Config;
 use shared::response;
@@ -15,41 +14,148 @@ mod cmd;
 mod constants;
 mod menu;
 mod rpc;
+mod window;
 
-fn _center_window(window: &Window) {
-    if let Some(monitor) = window.current_monitor().unwrap() {
-        let size = monitor.size();
-        let scale = monitor.scale_factor();
+fn main() {
+    let file_appender = tracing_appender::rolling::daily(Config::logs_dir(), "client.log");
+    let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
 
-        let middle = (size.width as f64 / (scale * 2.0)) - (constants::INPUT_WIDTH / 2.0);
+    tracing_subscriber::fmt()
+        .with_thread_names(true)
+        .with_writer(non_blocking)
+        .init();
 
-        window
-            .set_position(tauri::Position::Logical(tauri::LogicalPosition {
-                x: middle,
-                y: constants::INPUT_Y,
-            }))
-            .unwrap();
+    let ctx = tauri::generate_context!();
+
+    tauri::Builder::default()
+        .invoke_handler(tauri::generate_handler![
+            cmd::escape,
+            cmd::open_result,
+            cmd::search_docs,
+            cmd::search_lenses,
+            cmd::resize_window
+        ])
+        .menu(menu::get_app_menu())
+        .setup(|app| {
+            // hide from dock (also hides menu bar)
+            #[cfg(target_os = "macos")]
+            app.set_activation_policy(tauri::ActivationPolicy::Accessory);
+
+            // Only show in dev/debug mode.
+            #[cfg(debug_assertions)]
+            app.get_window("main").unwrap().open_devtools();
+
+            let window = app.get_window("main").unwrap();
+
+            // Start up backend (only in release mode)
+            #[cfg(not(debug_assertions))]
+            rpc::check_and_start_backend();
+
+            // Wait for the server to boot up
+            app.manage(tauri::async_runtime::block_on(rpc::RpcClient::new()));
+
+            // Register global shortcut
+            let mut shortcuts = app.global_shortcut_manager();
+            if !shortcuts.is_registered(constants::SHORTCUT).unwrap() {
+                let window = window.clone();
+                shortcuts
+                    .register(constants::SHORTCUT, move || {
+                        if window.is_visible().unwrap() {
+                            window::hide_window(&window);
+                        } else {
+                            window::show_window(&window);
+                        }
+                    })
+                    .unwrap();
+            }
+
+            // Center window horizontally in the current screen
+            window::center_window(&window);
+
+            Ok(())
+        })
+        .system_tray(SystemTray::new().with_menu(menu::get_tray_menu()))
+        .on_window_event(|event| {
+            if let tauri::WindowEvent::Focused(is_focused) = event.event() {
+                if !is_focused {
+                    let handle = event.window();
+                    window::hide_window(handle);
+                }
+            }
+        })
+        .on_system_tray_event(|app, event| match event {
+            SystemTrayEvent::LeftClick { .. } => update_tray_menu(app),
+            SystemTrayEvent::RightClick { .. } => update_tray_menu(app),
+            SystemTrayEvent::MenuItemClick { id, .. } => {
+                let item_handle = app.tray_handle().get_item(&id);
+                match id.as_str() {
+                    menu::CRAWL_STATUS_MENU_ITEM => {
+                        let rpc = app.state::<rpc::RpcClient>().inner();
+
+                        let is_paused = tauri::async_runtime::block_on(pause_crawler(rpc));
+                        let new_label = if is_paused {
+                            "▶️ Resume indexing"
+                        } else {
+                            "⏸ Pause indexing"
+                        };
+
+                        item_handle.set_title(new_label).unwrap();
+                    }
+                    menu::OPEN_LENSES_FOLDER => {
+                        open_folder(Config::lenses_dir());
+                    }
+                    menu::OPEN_SETTINGS_FOLDER => {
+                        open_folder(Config::prefs_dir());
+                    }
+                    menu::TOGGLE_MENU_ITEM => {
+                        let window = app.get_window("main").unwrap();
+                        let new_title = if window.is_visible().unwrap() {
+                            window::hide_window(&window);
+                            "Show"
+                        } else {
+                            window::show_window(&window);
+                            "Hide"
+                        };
+                        item_handle.set_title(new_title).unwrap();
+                    }
+                    menu::QUIT_MENU_ITEM => {
+                        app.exit(0);
+                    }
+                    _ => {}
+                }
+            }
+            _ => {}
+        })
+        .run(ctx)
+        .expect("error while running tauri application");
+}
+
+async fn app_status(rpc: &rpc::RpcClient) -> Option<response::AppStatus> {
+    match rpc
+        .client
+        .call_method::<Value, response::AppStatus>("app_stats", "", Value::Null)
+        .await
+    {
+        Ok(resp) => Some(resp),
+        Err(err) => {
+            log::error!("{}", err);
+            None
+        }
     }
 }
 
-fn _hide_window(window: &Window) {
-    window.hide().unwrap();
-    window.emit("clear_search", true).unwrap();
-}
-
-fn _show_window(window: &Window) {
-    window.show().unwrap();
-    window.set_focus().unwrap();
-    cmd::resize_window(window.clone(), constants::INPUT_HEIGHT);
-    _center_window(window);
-}
-
-#[allow(dead_code)]
-fn check_and_start_backend() {
-    let _ = Command::new_sidecar("spyglass-server")
-        .expect("failed to create `spyglass-server` binary command")
-        .spawn()
-        .expect("Failed to spawn sidecar");
+async fn pause_crawler(rpc: &rpc::RpcClient) -> bool {
+    match rpc
+        .client
+        .call_method::<Value, response::AppStatus>("toggle_pause", "", Value::Null)
+        .await
+    {
+        Ok(resp) => resp.is_paused,
+        Err(err) => {
+            log::error!("{}", err);
+            false
+        }
+    }
 }
 
 fn open_folder(folder: PathBuf) {
@@ -103,147 +209,5 @@ fn update_tray_menu(app: &AppHandle) {
                 app_status.num_queued.to_formatted_string(&Locale::en)
             ))
             .unwrap();
-    }
-}
-
-fn main() {
-    let file_appender = tracing_appender::rolling::daily(Config::logs_dir(), "client.log");
-    let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
-
-    tracing_subscriber::fmt()
-        .with_thread_names(true)
-        .with_writer(non_blocking)
-        .init();
-
-    let ctx = tauri::generate_context!();
-
-    tauri::Builder::default()
-        .invoke_handler(tauri::generate_handler![
-            cmd::escape,
-            cmd::open_result,
-            cmd::search_docs,
-            cmd::search_lenses,
-            cmd::resize_window
-        ])
-        .menu(menu::get_app_menu())
-        .setup(|app| {
-            // hide from dock (also hides menu bar)
-            #[cfg(target_os = "macos")]
-            app.set_activation_policy(tauri::ActivationPolicy::Accessory);
-
-            // Only show in dev/debug mode.
-            #[cfg(debug_assertions)]
-            app.get_window("main").unwrap().open_devtools();
-
-            let window = app.get_window("main").unwrap();
-
-            // Start up backend (only in release mode)
-            #[cfg(not(debug_assertions))]
-            check_and_start_backend();
-
-            // Wait for the server to boot up
-            app.manage(tauri::async_runtime::block_on(rpc::RpcClient::new()));
-
-            // Register global shortcut
-            let mut shortcuts = app.global_shortcut_manager();
-            if !shortcuts.is_registered(constants::SHORTCUT).unwrap() {
-                let window = window.clone();
-                shortcuts
-                    .register(constants::SHORTCUT, move || {
-                        if window.is_visible().unwrap() {
-                            _hide_window(&window);
-                        } else {
-                            _show_window(&window);
-                        }
-                    })
-                    .unwrap();
-            }
-
-            // Center window horizontally in the current screen
-            _center_window(&window);
-
-            Ok(())
-        })
-        .system_tray(SystemTray::new().with_menu(menu::get_tray_menu()))
-        .on_window_event(|event| {
-            if let tauri::WindowEvent::Focused(is_focused) = event.event() {
-                if !is_focused {
-                    let handle = event.window();
-                    _hide_window(handle);
-                }
-            }
-        })
-        .on_system_tray_event(|app, event| match event {
-            SystemTrayEvent::LeftClick { .. } => update_tray_menu(app),
-            SystemTrayEvent::RightClick { .. } => update_tray_menu(app),
-            SystemTrayEvent::MenuItemClick { id, .. } => {
-                let item_handle = app.tray_handle().get_item(&id);
-                match id.as_str() {
-                    menu::CRAWL_STATUS_MENU_ITEM => {
-                        let rpc = app.state::<rpc::RpcClient>().inner();
-
-                        let is_paused = tauri::async_runtime::block_on(pause_crawler(rpc));
-                        let new_label = if is_paused {
-                            "▶️ Resume indexing"
-                        } else {
-                            "⏸ Pause indexing"
-                        };
-
-                        item_handle.set_title(new_label).unwrap();
-                    }
-                    menu::OPEN_LENSES_FOLDER => {
-                        open_folder(Config::lenses_dir());
-                    }
-                    menu::OPEN_SETTINGS_FOLDER => {
-                        open_folder(Config::prefs_dir());
-                    }
-                    menu::TOGGLE_MENU_ITEM => {
-                        let window = app.get_window("main").unwrap();
-                        let new_title = if window.is_visible().unwrap() {
-                            _hide_window(&window);
-                            "Show"
-                        } else {
-                            _show_window(&window);
-                            "Hide"
-                        };
-                        item_handle.set_title(new_title).unwrap();
-                    }
-                    menu::QUIT_MENU_ITEM => {
-                        app.exit(0);
-                    }
-                    _ => {}
-                }
-            }
-            _ => {}
-        })
-        .run(ctx)
-        .expect("error while running tauri application");
-}
-
-async fn app_status(rpc: &rpc::RpcClient) -> Option<response::AppStatus> {
-    match rpc
-        .client
-        .call_method::<Value, response::AppStatus>("app_stats", "", Value::Null)
-        .await
-    {
-        Ok(resp) => Some(resp),
-        Err(err) => {
-            log::error!("{}", err);
-            None
-        }
-    }
-}
-
-async fn pause_crawler(rpc: &rpc::RpcClient) -> bool {
-    match rpc
-        .client
-        .call_method::<Value, response::AppStatus>("toggle_pause", "", Value::Null)
-        .await
-    {
-        Ok(resp) => resp.is_paused,
-        Err(err) => {
-            log::error!("{}", err);
-            false
-        }
     }
 }

--- a/crates/tauri/src/main.rs
+++ b/crates/tauri/src/main.rs
@@ -2,6 +2,7 @@
     all(not(debug_assertions), target_os = "windows"),
     windows_subsystem = "windows"
 )]
+use std::path::PathBuf;
 use jsonrpc_core::Value;
 use num_format::{Locale, ToFormattedString};
 use tauri::api::process::Command;
@@ -57,6 +58,27 @@ fn check_and_start_backend() {
         .expect("Failed to spawn sidecar");
 }
 
+
+fn open_folder(folder: PathBuf) {
+    #[cfg(target_os = "linux")]
+    std::process::Command::new("xdg-open")
+        .arg(folder)
+        .spawn()
+        .unwrap();
+
+    #[cfg(target_os = "macos")]
+    std::process::Command::new("open")
+        .arg(folder)
+        .spawn()
+        .unwrap();
+
+    #[cfg(target_os = "windows")]
+    std::process::Command::new("explorer")
+        .arg(folder)
+        .spawn()
+        .unwrap();
+
+}
 fn main() {
     let file_appender = tracing_appender::rolling::daily(Config::logs_dir(), "client.log");
     let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
@@ -174,16 +196,10 @@ fn main() {
                         item_handle.set_title(new_label).unwrap();
                     }
                     menu::OPEN_LENSES_FOLDER => {
-                        std::process::Command::new("open")
-                            .arg(Config::lenses_dir())
-                            .spawn()
-                            .unwrap();
+                        open_folder(Config::lenses_dir());
                     }
                     menu::OPEN_SETTINGS_FOLDER => {
-                        std::process::Command::new("open")
-                            .arg(Config::prefs_dir())
-                            .spawn()
-                            .unwrap();
+                        open_folder(Config::prefs_dir());
                     }
                     menu::TOGGLE_MENU_ITEM => {
                         let window = app.get_window("main").unwrap();

--- a/crates/tauri/src/main.rs
+++ b/crates/tauri/src/main.rs
@@ -2,12 +2,13 @@
     all(not(debug_assertions), target_os = "windows"),
     windows_subsystem = "windows"
 )]
-use std::path::PathBuf;
 use jsonrpc_core::Value;
 use num_format::{Locale, ToFormattedString};
+use std::path::PathBuf;
 use tauri::api::process::Command;
 use tauri::{
-    GlobalShortcutManager, LogicalSize, Manager, Size, State, SystemTray, SystemTrayEvent, Window,
+    AppHandle, GlobalShortcutManager, LogicalSize, Manager, Size, State, SystemTray,
+    SystemTrayEvent, Window,
 };
 
 use shared::config::Config;
@@ -58,7 +59,6 @@ fn check_and_start_backend() {
         .expect("Failed to spawn sidecar");
 }
 
-
 fn open_folder(folder: PathBuf) {
     #[cfg(target_os = "linux")]
     std::process::Command::new("xdg-open")
@@ -77,8 +77,42 @@ fn open_folder(folder: PathBuf) {
         .arg(folder)
         .spawn()
         .unwrap();
-
 }
+
+fn update_tray_menu(app: &AppHandle) {
+    let rpc = app.state::<rpc::RpcClient>().inner();
+
+    let app_status = tauri::async_runtime::block_on(app_status(rpc));
+    let handle = app.tray_handle();
+
+    if let Some(app_status) = app_status {
+        handle
+            .get_item(menu::CRAWL_STATUS_MENU_ITEM)
+            .set_title(if app_status.is_paused {
+                "▶️ Resume indexing"
+            } else {
+                "⏸ Pause indexing"
+            })
+            .unwrap();
+
+        handle
+            .get_item(menu::NUM_DOCS_MENU_ITEM)
+            .set_title(format!(
+                "{} documents indexed",
+                app_status.num_docs.to_formatted_string(&Locale::en)
+            ))
+            .unwrap();
+
+        handle
+            .get_item(menu::NUM_QUEUED_MENU_ITEM)
+            .set_title(format!(
+                "{} in queue",
+                app_status.num_queued.to_formatted_string(&Locale::en)
+            ))
+            .unwrap();
+    }
+}
+
 fn main() {
     let file_appender = tracing_appender::rolling::daily(Config::logs_dir(), "client.log");
     let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
@@ -147,39 +181,8 @@ fn main() {
             }
         })
         .on_system_tray_event(|app, event| match event {
-            SystemTrayEvent::LeftClick { .. } => {
-                let rpc = app.state::<rpc::RpcClient>().inner();
-
-                let app_status = tauri::async_runtime::block_on(app_status(rpc));
-                let handle = app.tray_handle();
-
-                if let Some(app_status) = app_status {
-                    handle
-                        .get_item(menu::CRAWL_STATUS_MENU_ITEM)
-                        .set_title(if app_status.is_paused {
-                            "▶️ Resume indexing"
-                        } else {
-                            "⏸ Pause indexing"
-                        })
-                        .unwrap();
-
-                    handle
-                        .get_item(menu::NUM_DOCS_MENU_ITEM)
-                        .set_title(format!(
-                            "{} documents indexed",
-                            app_status.num_docs.to_formatted_string(&Locale::en)
-                        ))
-                        .unwrap();
-
-                    handle
-                        .get_item(menu::NUM_QUEUED_MENU_ITEM)
-                        .set_title(format!(
-                            "{} in queue",
-                            app_status.num_queued.to_formatted_string(&Locale::en)
-                        ))
-                        .unwrap();
-                }
-            }
+            SystemTrayEvent::LeftClick { .. } => update_tray_menu(app),
+            SystemTrayEvent::RightClick { .. } => update_tray_menu(app),
             SystemTrayEvent::MenuItemClick { id, .. } => {
                 let item_handle = app.tray_handle().get_item(&id);
                 match id.as_str() {

--- a/crates/tauri/src/main.rs
+++ b/crates/tauri/src/main.rs
@@ -6,34 +6,27 @@ use jsonrpc_core::Value;
 use num_format::{Locale, ToFormattedString};
 use std::path::PathBuf;
 use tauri::api::process::Command;
-use tauri::{
-    AppHandle, GlobalShortcutManager, LogicalSize, Manager, Size, State, SystemTray,
-    SystemTrayEvent, Window,
-};
+use tauri::{AppHandle, GlobalShortcutManager, Manager, SystemTray, SystemTrayEvent, Window};
 
 use shared::config::Config;
-use shared::{request, response};
+use shared::response;
 
+mod cmd;
+mod constants;
 mod menu;
 mod rpc;
-
-const INPUT_WIDTH: f64 = 640.0;
-const INPUT_HEIGHT: f64 = 80.0;
-const INPUT_Y: f64 = 128.0;
-
-const SHORTCUT: &str = "CmdOrCtrl+Shift+/";
 
 fn _center_window(window: &Window) {
     if let Some(monitor) = window.current_monitor().unwrap() {
         let size = monitor.size();
         let scale = monitor.scale_factor();
 
-        let middle = (size.width as f64 / (scale * 2.0)) - (INPUT_WIDTH / 2.0);
+        let middle = (size.width as f64 / (scale * 2.0)) - (constants::INPUT_WIDTH / 2.0);
 
         window
             .set_position(tauri::Position::Logical(tauri::LogicalPosition {
                 x: middle,
-                y: INPUT_Y,
+                y: constants::INPUT_Y,
             }))
             .unwrap();
     }
@@ -47,7 +40,7 @@ fn _hide_window(window: &Window) {
 fn _show_window(window: &Window) {
     window.show().unwrap();
     window.set_focus().unwrap();
-    resize_window(window.clone(), INPUT_HEIGHT);
+    cmd::resize_window(window.clone(), constants::INPUT_HEIGHT);
     _center_window(window);
 }
 
@@ -126,11 +119,11 @@ fn main() {
 
     tauri::Builder::default()
         .invoke_handler(tauri::generate_handler![
-            escape,
-            open_result,
-            search_docs,
-            search_lenses,
-            resize_window
+            cmd::escape,
+            cmd::open_result,
+            cmd::search_docs,
+            cmd::search_lenses,
+            cmd::resize_window
         ])
         .menu(menu::get_app_menu())
         .setup(|app| {
@@ -153,10 +146,10 @@ fn main() {
 
             // Register global shortcut
             let mut shortcuts = app.global_shortcut_manager();
-            if !shortcuts.is_registered(SHORTCUT).unwrap() {
+            if !shortcuts.is_registered(constants::SHORTCUT).unwrap() {
                 let window = window.clone();
                 shortcuts
-                    .register(SHORTCUT, move || {
+                    .register(constants::SHORTCUT, move || {
                         if window.is_visible().unwrap() {
                             _hide_window(&window);
                         } else {
@@ -227,28 +220,6 @@ fn main() {
         .expect("error while running tauri application");
 }
 
-#[tauri::command]
-async fn escape(window: tauri::Window) -> Result<(), String> {
-    _hide_window(&window);
-    Ok(())
-}
-
-#[tauri::command]
-async fn open_result(_: tauri::Window, url: &str) -> Result<(), String> {
-    open::that(url).unwrap();
-    Ok(())
-}
-
-#[tauri::command]
-fn resize_window(window: tauri::Window, height: f64) {
-    window
-        .set_size(Size::Logical(LogicalSize {
-            width: INPUT_WIDTH,
-            height,
-        }))
-        .unwrap();
-}
-
 async fn app_status(rpc: &rpc::RpcClient) -> Option<response::AppStatus> {
     match rpc
         .client
@@ -273,58 +244,6 @@ async fn pause_crawler(rpc: &rpc::RpcClient) -> bool {
         Err(err) => {
             log::error!("{}", err);
             false
-        }
-    }
-}
-
-#[tauri::command]
-async fn search_docs<'r>(
-    _: tauri::Window,
-    rpc: State<'r, rpc::RpcClient>,
-    lenses: Vec<String>,
-    query: &str,
-) -> Result<Vec<response::SearchResult>, String> {
-    let data = request::SearchParam {
-        lenses,
-        query: query.to_string(),
-    };
-
-    match rpc
-        .client
-        .call_method::<(request::SearchParam,), response::SearchResults>("search_docs", "", (data,))
-        .await
-    {
-        Ok(resp) => Ok(resp.results.to_vec()),
-        Err(err) => {
-            log::error!("{}", err);
-            Ok(Vec::new())
-        }
-    }
-}
-
-#[tauri::command]
-async fn search_lenses<'r>(
-    _: tauri::Window,
-    rpc: State<'r, rpc::RpcClient>,
-    query: &str,
-) -> Result<Vec<response::LensResult>, String> {
-    let data = request::SearchLensesParam {
-        query: query.to_string(),
-    };
-
-    match rpc
-        .client
-        .call_method::<(request::SearchLensesParam,), response::SearchLensesResp>(
-            "search_lenses",
-            "",
-            (data,),
-        )
-        .await
-    {
-        Ok(resp) => Ok(resp.results.to_vec()),
-        Err(err) => {
-            log::error!("{}", err);
-            Ok(Vec::new())
         }
     }
 }

--- a/crates/tauri/src/rpc.rs
+++ b/crates/tauri/src/rpc.rs
@@ -1,11 +1,20 @@
 use jsonrpc_core_client::{transports::ipc, TypedClient};
 use shared::rpc::gen_ipc_path;
+use tauri::api::process::Command;
 use tokio_retry::strategy::{jitter, ExponentialBackoff};
 use tokio_retry::Retry;
 
 pub struct RpcClient {
     pub client: TypedClient,
     pub endpoint: String,
+}
+
+#[allow(dead_code)]
+pub fn check_and_start_backend() {
+    let _ = Command::new_sidecar("spyglass-server")
+        .expect("failed to create `spyglass-server` binary command")
+        .spawn()
+        .expect("Failed to spawn sidecar");
 }
 
 async fn connect(endpoint: String) -> Result<TypedClient, ()> {

--- a/crates/tauri/src/rpc.rs
+++ b/crates/tauri/src/rpc.rs
@@ -1,7 +1,7 @@
 use jsonrpc_core_client::{transports::ipc, TypedClient};
 use shared::rpc::gen_ipc_path;
+use tokio_retry::strategy::{jitter, ExponentialBackoff};
 use tokio_retry::Retry;
-use tokio_retry::strategy::{ExponentialBackoff, jitter};
 
 pub struct RpcClient {
     pub client: TypedClient,
@@ -24,11 +24,9 @@ impl RpcClient {
             .map(jitter) // add jitter to delays
             .take(3);
 
-
-        let client: TypedClient = Retry::spawn(
-            retry_strategy,
-            || { connect(endpoint.clone()) }
-        ).await.unwrap();
+        let client: TypedClient = Retry::spawn(retry_strategy, || connect(endpoint.clone()))
+            .await
+            .unwrap();
 
         RpcClient {
             client,

--- a/crates/tauri/src/window.rs
+++ b/crates/tauri/src/window.rs
@@ -1,0 +1,40 @@
+use tauri::{LogicalSize, Size, Window};
+
+use crate::{cmd, constants};
+
+pub fn center_window(window: &Window) {
+    if let Some(monitor) = window.current_monitor().unwrap() {
+        let size = monitor.size();
+        let scale = monitor.scale_factor();
+
+        let middle = (size.width as f64 / (scale * 2.0)) - (constants::INPUT_WIDTH / 2.0);
+
+        window
+            .set_position(tauri::Position::Logical(tauri::LogicalPosition {
+                x: middle,
+                y: constants::INPUT_Y,
+            }))
+            .unwrap();
+    }
+}
+
+pub fn hide_window(window: &Window) {
+    window.hide().unwrap();
+    window.emit("clear_search", true).unwrap();
+}
+
+pub fn resize_window(window: &Window, height: f64) {
+    window
+        .set_size(Size::Logical(LogicalSize {
+            width: constants::INPUT_WIDTH,
+            height,
+        }))
+        .unwrap();
+}
+
+pub fn show_window(window: &Window) {
+    window.show().unwrap();
+    window.set_focus().unwrap();
+    cmd::resize_window(window.clone(), constants::INPUT_HEIGHT);
+    center_window(window);
+}

--- a/crates/tauri/tauri.conf.json
+++ b/crates/tauri/tauri.conf.json
@@ -29,13 +29,11 @@
       "shortDescription": "",
       "longDescription": "",
       "deb": {
-        "depends": [],
-        "useBootstrapper": false
+        "depends": []
       },
       "macOS": {
         "frameworks": [],
         "minimumSystemVersion": "",
-        "useBootstrapper": false,
         "exceptionDomain": "",
         "signingIdentity": null,
         "entitlements": null

